e2e/credentials-server.e2e.spec.ts

```diff
@@ -0,0 +1,187 @@
+
+    return !result.error;
+  } catch (err: any) {
+    return false;
+  }
+}
+
+const describeWithCreds = haveCredentialServerHelper() ? test.describe : test.skip;
+
+describeWithCreds('Credentials server', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+  let serverState: ServerState;
+  let authString: string;
+  let page: Page;
+  const appPath = path.join(__dirname, '../');
+  // Assign these values on first request once we have an authString
+  // And we can't assign to ipaddr on Windows here because we need an async context.
+  let command = '';
+  let ipaddr: string|undefined = '';
+  let initialArgs: string[] = [];
+
+  async function doRequest(path: string, body = '') {
+    if (!ipaddr) {
+      if (os.platform() === 'win32') {
+        command = 'wsl';
+        ipaddr = await wslHostIPv4Address();
+        if (!ipaddr) {
+          throw new Error('Failed to get the WSL IP address');
+        }
+        initialArgs = ['--distribution', 'rancher-desktop', '--exec', 'curl'];
+      } else {
+        command = 'curl';
+        ipaddr = 'localhost';
+      }
+      initialArgs = initialArgs.concat(['--silent', '--user', authString, '--request', 'POST']);
+    }
+    const args = initialArgs.concat([`http://${ ipaddr }:${ serverState.port }/${ path }`]);
+
+    if (body.length) {
+      args.push('--data', body);
+    }
+    const { stdout, stderr } = await spawnFile(command, args, { stdio: 'pipe' });
+
+    expect(stderr).toEqual('');
+
+    return stdout;
+  }
+
+  test.describe.configure({ mode: 'serial' });
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+    electronApp = await _electron.launch({
+      args: [
+        appPath,
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({
+      screenshots: true,
+      snapshots:   true
+    });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: playwrightReportAssets(path.basename(__filename)) });
+    await electronApp.close();
+  });
+
+  test('should emit connection information', async() => {
+    const dataPath = path.join(paths.appHome, 'credential-server.json');
+    const dataRaw = await fs.promises.readFile(dataPath, 'utf-8');
+
+    serverState = JSON.parse(dataRaw);
+    expect(typeof serverState.user).toBe('string');
+    expect(typeof serverState.password).toBe('string');
+    expect(typeof serverState.port).toBe('number');
+    expect(typeof serverState.pid).toBe('number');
+    authString = `${ serverState.user }:${ serverState.password }`;
+  });
+
+  test('should require authentication', async() => {
+    const url = `http://127.0.0.1:${ serverState.port }/list`;
```

Comment:
This doesn't seem to work on Windows (because it's listening on the WSL address).

Reply:
True, I fixed this on branch 2166 but I'll pull it in here.